### PR TITLE
Add PostgreSQL JDBC driver jar to Iceberg image

### PR DESCRIPTION
**Pull request by ebyhr**
Relates to https://github.com/trinodb/trino/pull/11772

### CODE REVIEW

File: testing/spark3.0-iceberg/Dockerfile [hunk 1]
```diff
@@ -33,6 +33,9 @@ WORKDIR ${SPARK_HOME}/jars
 # install Iceberg
 RUN wget -nv "https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-spark-runtime-${ICEBERG_JAR_VERSION}/${ICEBERG_VERSION}/iceberg-spark-runtime-${ICEBERG_JAR_VERSION}-${ICEBERG_VERSION}.jar"
 
+# install PostgreSQL driver for JDBC catalog
+RUN wget -nv --no-check-certificate "https://jdbc.postgresql.org/download/postgresql-42.3.5.jar"
```
Comment (findepi) [inline testing/spark3.0-iceberg/Dockerfile:37]: Why do we need `--no-check-certificate`? (we don't need it above; also, well, it's not secure)

Comment (ebyhr) [reply testing/spark3.0-iceberg/Dockerfile]: I faced the below error when building the image. Switched to maven now.
```
Step 11/14 : RUN wget -nv --no-check-certificate "https://jdbc.postgresql.org/download/postgresql-42.3.5.jar"
 ---> Running in 5755549870dd
WARNING: cannot verify jdbc.postgresql.org's certificate, issued by ‘/C=US/O=Let's Encrypt/CN=R3’:
  Issued certificate has expired.
2022-05-06 06:24:27 URL:https://jdbc.postgresql.org/download/postgresql-42.3.5.jar [1041066/1041066] -> "postgresql-42.3.5.jar" [1]
```

Comment (findepi) [reply testing/spark3.0-iceberg/Dockerfile]: Interesting. Both my browser and `wget` running locally had no problem downloading `https://jdbc.postgresql.org/download/postgresql-42.3.5.jar`.

Comment (ebyhr) [reply testing/spark3.0-iceberg/Dockerfile]: Me too. Another option is installing `ca-certificates` beforehand. 
```
yum install ca-certificates
```

Comment (findepi) [reply testing/spark3.0-iceberg/Dockerfile]: installing `ca-certificates` is definitely better than `--no-check-certificate`

**Review by findepi — APPROVED** [review]

**Pull request merged**